sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -248,57 +244,91 @@ private[sql] trait SQLTestUtilsBase
           !spark.sessionState.catalog.functionExists(FunctionIdentifier(functionName)),
           s"Function $functionName should have been dropped. But, it still exists.")
       }
-    }
+    )
   }
 
   /**
    * Drops temporary view `viewNames` after calling `f`.
    */
   protected def withTempView(viewNames: String*)(f: => Unit): Unit = {
-    try f finally {
-      // If the test failed part way, we don't want to mask the failure by failing to remove
-      // temp views that never got created.
-      try viewNames.foreach(spark.catalog.dropTempView) catch {
-        case _: NoSuchTableException =>
-      }
-    }
+    tryWithFinallyBlock(f)(viewNames.foreach(spark.catalog.dropTempView))
   }
 
   /**
    * Drops global temporary view `viewNames` after calling `f`.
    */
   protected def withGlobalTempView(viewNames: String*)(f: => Unit): Unit = {
-    try f finally {
-      // If the test failed part way, we don't want to mask the failure by failing to remove
-      // global temp views that never got created.
-      try viewNames.foreach(spark.catalog.dropGlobalTempView) catch {
-        case _: NoSuchTableException =>
-      }
-    }
+    tryWithFinallyBlock(f)(viewNames.foreach(spark.catalog.dropGlobalTempView))
   }
 
   /**
    * Drops table `tableName` after calling `f`.
    */
   protected def withTable(tableNames: String*)(f: => Unit): Unit = {
-    try f finally {
+    tryWithFinallyBlock(f)(
       tableNames.foreach { name =>
         spark.sql(s"DROP TABLE IF EXISTS $name")
       }
-    }
+    )
   }
 
   /**
    * Drops view `viewName` after calling `f`.
    */
   protected def withView(viewNames: String*)(f: => Unit): Unit = {
-    try f finally {
+    tryWithFinallyBlock(f)(
       viewNames.foreach { name =>
         spark.sql(s"DROP VIEW IF EXISTS $name")
       }
+    )
+  }
+
+  /**
+   * Drops cache `cacheName` after calling `f`.
+   */
+  protected def withCache(cacheNames: String*)(f: => Unit): Unit = {
+    tryWithFinallyBlock(f)(cacheNames.foreach(uncacheTable))
+  }
+
+  /**
+   * Executes the given tryBlock and then the given finallyBlock no matter whether tryBlock throws
+   * an exception. If both tryBlock and finallyBlock throw exceptions, the exception thrown
+   * from the finallyBlock with be added to the exception thrown from tryBlock as a
+   * suppress exception. It helps to avoid masking the exception from tryBlock with exception
+   * from finallyBlock
+   */
+  private def tryWithFinallyBlock(tryBlock: => Unit)(finallyBlock: => Unit): Unit = {
```

Comment:
Could you implement `withCache` simply like the other `withXXX` functions instead of introducing `tryWithFinallyBlock` in this PR?

Reply:
You can do that kind of refactoring later in another PR.

Reply:
Many thanks for keeping review this PR and also your comments. I reverted the implementation of 'withCache' and I will submit another PR for 'tryWithFinallyBlock'.